sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeCreator.scala

```diff
@@ -106,7 +106,23 @@ case class GpuCreateMap(children: Seq[Expression], useStringTypeWhenEmpty: Boole
       val structs = Range(0, columns.length, 2)
         .safeMap(i => ColumnVector.makeStruct(columns(i), columns(i + 1)))
       withResource(structs) { _ =>
-        GpuColumnVector.from(ColumnVector.makeList(numRows, DType.STRUCT, structs: _*), dataType)
+        withResource(ColumnVector.makeList(numRows, DType.STRUCT, structs: _*)) { listOfStruct =>
+          withResource(listOfStruct.dropListDuplicatesWithKeysValues()) { deduped =>
+            if (SQLConf.get.getConf(SQLConf.MAP_KEY_DEDUP_POLICY) ==
+                SQLConf.MapKeyDedupPolicy.LAST_WIN.toString) {
+              GpuColumnVector.from(deduped.incRefCount(), dataType)
+            } else {
+              withResource(deduped.getChildColumnView(0)) { a =>
+                withResource(listOfStruct.getChildColumnView(0)) { b =>
+                  if (a.getRowCount != b.getRowCount) {
+                    throw GpuMapUtils.duplicateMapKeyFoundError
+                  }
+                }
+              }
+              GpuColumnVector.from(deduped.incRefCount(), dataType)
```

Comment:
So are we going to have the same output expr for both if/else cases?

Reply:
Yes. The `MAP_KEY_DEDUP_POLICY` config is either `LAST_WIN` or `EXCEPTION`. For ` LAST_WIN` we return the deduped data. For `EXCEPTION` we throw an exception if there are duplicates but if there are no duplicates we return the original map, which is the same as the deduped map, so we can return either.